client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/multimap/operations/MultimapOperationsFactory.java

```diff
@@ -0,0 +1,119 @@
+package org.infinispan.client.hotrod.impl.multimap.operations;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.infinispan.client.hotrod.Flag;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.Configuration;
+import org.infinispan.client.hotrod.impl.operations.HotRodOperation;
+import org.infinispan.client.hotrod.impl.protocol.Codec;
+import org.infinispan.client.hotrod.impl.transport.TransportFactory;
+
+import net.jcip.annotations.Immutable;
+
+/**
+ * Factory for {@link HotRodOperation} objects on Multimap.
+ *
+ * @author karesti@redhat.com
+ * @since 9.2
+ */
+@Immutable
+public class MultimapOperationsFactory {
+
+   private final ThreadLocal<Integer> flagsMap = new ThreadLocal<>();
```

Comment:
I don't see any usage of the thread local. IMO, it can be removed.

Reply:
Flags are passed to the operations through them ...

Reply:
but I didn't find any `flagsMap.set()`. its value will be always zero (or null)

Reply:
you are right, I didn't realise before. I'm going to just rework all the flag stuff in the issue I opened after will's commets on this : https://issues.jboss.org/browse/ISPN-8580